pandas/core/missing.py

```diff
@@ -356,12 +369,23 @@ def _interpolate_scipy_wrapper(
             y = y.copy()
         if not new_x.flags.writeable:
             new_x = new_x.copy()
-        method = alt_methods[method]
-        new_y = method(x, y, new_x, **kwargs)
+
+        if isinstance(method, str):
+            alt_method = alt_methods[method]
+            new_y = alt_method(x, y, new_x, **kwargs)
+        else:
+            raise ValueError(f"{method} is not a valid interp method")
```

Comment:
if non-str is invalid, shouldnt the annotation be `str` instead of `Optional[str]`?

Reply:
good catch (here and in `interpolate_1d`)